### PR TITLE
chore: Centralize sauce browsers lists

### PR DESCRIPTION
**Pull request by patrickhousley**
Moving the browsers list json files and sauce browsers retrieval script to a central location where they can be used by jil and wdio instead of each having their own version.
----

### Overview

Moved the script that generates the browser list json files as well as those json files into a shared location where JIL and WDIO could use them. The supported browsers list was getting duplicate entries in each of the category arrays so I modified the script so the `getBrowsers` function was stateless.

### Related Issue(s)

Extracted from #464

### CODE REVIEW

**Review by bjfield — APPROVED** [review]

File: tools/browsers-lists/browsers-supported.json [hunk 1]
```diff
@@ -2,107 +2,117 @@
...
       "browserName": "firefox",
+      "platformName": "Windows 10",
       "platform": "Windows 10",
-      "version": "109"
+      "version": "106",
+      "browserVersion": "106"
     },
     {
       "browserName": "firefox",
+      "platformName": "Windows 10",
```
Comment (metal-messiah) [inline tools/browsers-lists/browsers-supported.json:79]: These have extra properties now.  Is there an issue with doing this?

File: tools/browsers-lists/browsers-supported.json [hunk 2]
```diff
@@ -2,107 +2,117 @@
...
+      "browserVersion": "112"
+    }
+  ],
+  "safari": [
+    {
+      "browserName": "safari",
+      "platformName": "Mac 12",
+      "platform": "Mac 12",
+      "version": "15",
+      "browserVersion": "15"
```
Comment (metal-messiah) [inline tools/browsers-lists/browsers-supported.json:98]: Also missing insecureCerts property in safari desktop

File: tools/browsers-lists/browsers-supported.json (hunk 1 above)
Comment (patrickhousley) [reply tools/browsers-lists/browsers-supported.json:79]: It is accounted for in the change to `tools/jil/runner/index.js`

File: tools/browsers-lists/browsers-supported.json (hunk 2 above)
Comment (patrickhousley) [reply tools/browsers-lists/browsers-supported.json]: fixed, thanks for catching

File: tools/browsers-lists/browsers-supported.json [hunk 3]
```diff
@@ -113,59 +123,55 @@
...
+      "platform": "iOS",
+      "version": "15.4",
+      "device": "iPhone XS",
+      "appium:deviceName": "iPhone XS Simulator",
+      "appium:platformVersion": "15.4",
       "appium:automationName": "XCUITest",
       "sauce:options": {
         "appiumVersion": "1.22.3"
-      },
-      "acceptInsecureCerts": false
```
[the remark this answers is not in the document]
Comment (patrickhousley) [reply tools/browsers-lists/browsers-supported.json]: fixed, thanks for catching

**Review by metal-messiah — APPROVED** [review]

**Pull request merged**